Kickstarter-iOS/Views/Cells/PledgeAmountCell.swift

```diff
@@ -67,11 +67,15 @@ final class PledgeAmountCell: UITableViewCell, ValueCell {
 
   // MARK: - Configuration
 
-  func configureWith(value: (amount: Double, currency: String)) {
+  func configureWith(value: PledgeDataSource.PledgeInputRow) {
+    guard case .pledgeAmount(let amount, let currency) = value else {
```

Comment:
Should it be the responsibility of the cell to have this condition? Shouldn't the cell simply get data and do whatever it wants with it? I wonder if it should be the call site that has this condition not the "dumb" view layer that's not concerned about it at all (unless it has a VM)? Not sure if we can better destructure the data in the data source before we call `configureWith(value:)` on this cell 🤔

Reply:
Or maybe `PledgeDataSource.PledgeInputRow` should be able to provide optional values for all cases?

```swift
PledgeInputRow {
  case1
  case2
  ...

  // optional or non-optional depending on what's better for our use case
  var amount: String? { .. } // switch over
  var currency: String? { .. } // switch over
}
```

then we would hopefully not need to use the guard and can simply configure `AmountInputView` with these optionals (I guess) ... we can simply change the `configureWith` to take optionals since the internal logic of that on the `AmountInputView` would be fine with optionals

Reply:
Since we've decided to add individual view models to the input cells, I think we should pipe in the value of `PledgeDataSource.PledgeInputRow` directly into the cell's view model and allow it to check that the `enum` case is the correct one for the cell's needs.